RELEASE/scripts/autoscend/auto_util.ash

```diff
@@ -2696,189 +2696,304 @@ boolean acquireCombatMods(int amt, boolean doEquips)
 }
 
 
-boolean providePlusCombat(int amt)
-{
-	return providePlusCombat(amt, true);
-}
-boolean providePlusNonCombat(int amt)
-{
-	return providePlusNonCombat(amt, true);
-}
+float providePlusCombat(int amt, boolean doEquips, boolean speculative) {
+	auto_log_info((speculative ? "Checking if we can" : "Trying to") + " provide " + amt + " positive combat rate, " + (doEquips ? "with" : "without") + " equipment", "blue");
 
-boolean providePlusCombat(int amt, boolean doEquips)
-{
-	if(amt == 0)
-	{
-		return true;
-	}
-	boolean forbid_extra = true;
-	if(amt >= 25)
-	{
-		forbid_extra = false;		//do we want to stop when there are FREE methods to raise it further
+	float alreadyHave = numeric_modifier("Combat Rate");
+	float need = amt - alreadyHave;
+
+	if (need > 0) {
+		auto_log_debug("We currently have " + alreadyHave + ", so we need an extra " + need);
+	} else {
+		auto_log_debug("We already have enough!");
 	}
-	
-	//equipment handling
-	if(doEquips)
-	{
-		//to prevent flipflop resulting in failing 50% of the turns we always do this first. see issue #400
-		addToMaximize("200combat " + to_string(amt) + "max");
+
+	float delta = 0;
+
+	float result() {
+		return numeric_modifier("Combat Rate") + delta;
 	}
-	simMaximize();	//we do not want to repeatedly simulate equipment. simulate it once only.
-	int equipDiff = to_int(simValue("Combat Rate") - numeric_modifier("Combat Rate"));
 
-	boolean are_we_done()
-	{
-		if(numeric_modifier("Combat Rate").to_int() + equipDiff >= amt)
-		{
-			return true;
+	if (doEquips) {
+		string max = "200combat " + amt + "max";
+		if (speculative) {
+			simMaximizeWith(max);
+		} else {
+			addToMaximize("200combat " + amt + "max");
```

Comment:
shouldn't this be `addToMaximize(max);`
because I am not seeing string max being used anywhere and this would be the place to use it

Reply:
LoL that's a copy paste fail. I'll fix it (and the same code in provideInitiative).

Good spot.

Reply:
oh, also in `boolean providePlusNonCombat`